src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -906,6 +913,33 @@ private void StartMusic()
         Jukebox.Instance.Resume();
     }
 
+    /// <summary>
+    ///   Updates the arrowPosition variable to the top most point of the middle 3 rows
+    ///   Should be called on any layout change
+    /// </summary>
+    private void UpdateArrow()
+    {
+        // 1. Retrieves all the hexes in the microbe
+        // 2. Fetches only the middle 3 rows
+        // 3. Maps the R position to the actual height
+        // 4. Retrieves the minimum value, which is the top most location in the middle 3 rows
+        // The calculation falls back to 0 if there are no hexes found in the middle 3 rows
+        var highestPointInMiddleRows = editedMicrobeOrganelles
+            .SelectMany(organelle => organelle.Definition.Hexes.Select(hex => hex + organelle.Position))
+            .Where(hex => hex.Q >= -1 && hex.Q <= 1)
+            .Select(hex => hex.Q switch
+            {
+                -1 => hex.R - 1,
+                0 => hex.R - .5f,
+                _ => hex.R,
+            }).DefaultIfEmpty(0).Min();
```

Comment:
Even though this is commented, I think it would be much nicer to rewrite this with a foreach because this crosses the my threshold  of how much stuff should be allowed in a single LINQ statment